.github/workflows/release-e2e.yaml

```diff
@@ -0,0 +1,49 @@
+---
+name: release-e2e
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  release-e2e-kind:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '~1.18'
+
+    - name: Start a kind cluster
+      run: |
+        make kind-cluster
+
+    - name: Download kubectl
+      run: |
+        curl -LO "https://dl.k8s.io/release/${{env.KUBECTL_VERSION}}/bin/linux/amd64/kubectl"
```

Comment:
I realize we're not using a Makefile target to download kubectl so the `export KUBECTL_VERSION=...` in the Makefile won't propagate to github's environment context. I think we do the following instead:

- Remove the exported variable from the Makefile
- Update this command to inline a shell command: `curl -LO "https://dl.k8s.io/release/$(go list -m k8s.io/kubectl | cut -d" " -f2 | sed 's/^v0/v1/')/bin/linux/amd64/kubectl"`